src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -48,6 +48,12 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <TargetRuntime>Managed</TargetRuntime>
   </PropertyGroup>
 
+    <!-- Store values of certain intrinsic functions that won't change throughout the build. -->
+  <PropertyGroup Condition="'$(TargetFramework)' != ''">
+    <TargetFrameworkIdentifier Condition="'$(TargetFrameworkIdentifier)' == ''">$([MSBuild]::GetTargetFrameworkIdentifier($(TargetFramework)))</TargetFrameworkIdentifier>
+    <TargetFrameworkVersion Condition="'$(TargetFrameworkVersion)' == ''">$([MSBuild]::GetTargetFrameworkVersion($(TargetFramework)))</TargetFrameworkIdentifier>
```

Comment:
This worries me. Does it really not conflict with something NuGet does?

Reply:
I had a general concern overriding these, so I added the check to see if it didn't already have a value. Do you think this could mess with something nuget does where they do a similar "set it if it isn't already set" check?

When setting these values here, I considered prefixing this with an underscore for safety. Maybe we have a different naming convention for "Values that likely won't change during a build that we store for msbuild specific purposes"?